src/char_mapping.rs

```diff
@@ -6,6 +6,49 @@ use std::collections::HashMap;
 
 use {GlyphID, Result, Tape, Value};
 
+/// A mapping from a character code to a glyph id
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Mapping {
+    U8(HashMap<u8, GlyphID>),
+    U16(HashMap<u16, GlyphID>),
+    U32(HashMap<u32, GlyphID>),
+    None,
+}
+
+impl Mapping {
```

Comment:
Let’s not entwine struct definitions with implementations. All definitions come first; then all implementations follow in the same order.